tests/failedData.test.ts

```diff
@@ -0,0 +1,34 @@
+import { describe, it, expect, beforeAll, afterAll, afterEach } from "vitest";
+import { http, HttpResponse } from "msw";
+import { setupServer } from "msw/node";
+import { fetchData } from "@/lib/api";
+
+const apiUrl = process.env.PROMOCHATOR_API ?? "http://localhost:3000";
+
+// mocked api 500 response
+const server = setupServer(
+  http.post(`${apiUrl}/recommend/invoke`, () => {
+    return new HttpResponse("Internal Server Error", {
+      status: 500,
+    });
+  }),
+);
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
```

Comment:
to akurat mogło pójść do pliku z setupem

a jak byś potrzebował, żeby request dochodził do prawdziwego backendu to możesz użyć: https://mswjs.io/docs/api/passthrough